src/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/GaugeWave.cpp

```diff
@@ -0,0 +1,151 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/AnalyticSolutions/GeneralRelativity/GaugeWave.hpp"
+
+#include <cmath>  // IWYU pragma: keep
+#include <cstdlib>
+#include <numeric>
+#include <ostream>
+#include <pup.h>
+#include <utility>
+
+#include "DataStructures/DataBox/Prefixes.hpp"  // IWYU pragma: keep
+#include "DataStructures/DataVector.hpp"        // IWYU pragma: keep
+#include "DataStructures/TempBuffer.hpp"
+#include "DataStructures/Tensor/EagerMath/DeterminantAndInverse.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"  // IWYU pragma: keep
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Parallel/PupStlCpp11.hpp"
+#include "PointwiseFunctions/GeneralRelativity/ComputeSpacetimeQuantities.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/ContainerHelpers.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "Utilities/StdArrayHelpers.hpp"
+#include "Utilities/StdHelpers.hpp"
+#include "Utilities/TMPL.hpp"
+
+// IWYU pragma: no_include <complex>
+
+// IWYU pragma: no_forward_declare Tensor
+
+/// \cond
+namespace {
+// compute gauge wave H
+template <typename DataType>
+Scalar<DataType> gauge_wave_h(const tnsr::I<DataType, 3>& x, const double t,
+                              const double amplitude, const double wavelength) {
+  auto h = make_with_value<Scalar<DataType>>(x, 0.0);
+  get(h) =
+      -1.0 * amplitude * sin((get<0>(x) - t) * (2.0 * M_PI / wavelength)) + 1.0;
+  return h;
+}
+
+// compute gauge wave derivH: \partial_x H
+template <typename DataType>
+Scalar<DataType> gauge_wave_deriv_h(const tnsr::I<DataType, 3>& x,
+                                    const double t, const double amplitude,
+                                    const double wavelength) {
+  auto h = make_with_value<Scalar<DataType>>(x, 0.0);
+  get(h) = -1.0 * amplitude * (2.0 * M_PI / wavelength) *
+           cos((get<0>(x) - t) * (2.0 * M_PI / wavelength));
+  return h;
+}
+}  // namespace
+
+namespace gr {
+namespace Solutions {
+
+GaugeWave::GaugeWave(const double amplitude, const double wavelength,
+                     const OptionContext& context)
+    : amplitude_(amplitude), wavelength_(wavelength) {
+  if (amplitude_ < 0.0) {
+    PARSE_ERROR(context, "Amplitude must be non-negative. Given amplitude: "
+                             << amplitude_);
+  }
+  if (wavelength_ < 0.0) {
+    PARSE_ERROR(context, "Wavelength must be non-negative. Given wavelength: "
+                             << wavelength_);
+  }
+}
+
+void GaugeWave::pup(PUP::er& p) noexcept {
+  p | amplitude_;
+  p | wavelength_;
+}
+
+template <typename DataType>
+tuples::tagged_tuple_from_typelist<GaugeWave::tags<DataType>>
```

Comment:
I'll look through this once you've factored the code to be more like `Minkowski` and less like `KerrSchild`